src/s2wasm.h

```diff
@@ -96,6 +112,23 @@ class S2WasmBuilder {
 
   // utilities
 
+  // For fatal errors which could arise from input (i.e. not assertion failures)
+  class Fatal {
+   public:
+    Fatal() {
+      std::cerr << "Fatal: ";
+    }
+    template<typename T>
+    Fatal &operator<<(T arg) {
```

Comment:
Yes, although since I just ended up using std::ostream's << operator the forward is pointless anyway. I'll remove it.